apps/frontend/src/views/Checklist.vue

```diff
@@ -0,0 +1,313 @@
+      data: string;
+    };
+    const checklist = this.getChecklist(this.fileFilter);
+    if (checklist) {
+      const checklistString = ChecklistConverter.toChecklist(checklist);
+      const file: FileData[] = [
+        {
+          filename: checklist.filename,
+          data: '<?xml version="1.0" encoding="UTF-8"?>' + checklistString
+        }
+      ];
+      saveSingleOrMultipleFiles(file, 'ckl');
+    }
+  }
+
+  get selectedRule() {
+    let stillExists = false;
+    // Checks to see if the selected rule still exists after filtering
+    this.rules.forEach((item) => {
+      if (_.isEqual(FilteredDataModule.selectedRule, item)) {
+        stillExists = true;
+      }
+    });
+    if (stillExists) {
+      return FilteredDataModule.selectedRule;
+    }
+    return this.rules[0] ?? FilteredDataModule.emptyRule;
+  }
+
+  /**
+   * The current search terms, as modeled by the search bar
+   */
+  get searchTerm(): string {
+    return SearchModule.searchTerm;
+  }
+
+  set searchTerm(term: string) {
+    SearchModule.updateSearch(term);
+  }
+
+  /**
+   * The currently selected file, if one exists.
+   * Controlled by router.
+   */
+  get fileFilter(): FileID[] {
+    return FilteredDataModule.selectedChecklistIds;
+  }
+
+  getChecklist(fileID: FileID[]) {
+    return InspecDataModule.allChecklistFiles.find(
+      (f) => f.uniqueId === fileID[0]
+    );
+  }
+
+  /**
+   * The filter for charts. Contains all of our filter stuff
+   */
+  get allFilter(): ChecklistFilter {
+    return {
+      status: SearchModule.inFileSearchTerms.statusFilter,
+      severity: SearchModule.inFileSearchTerms.severityFilter,
+      fromFile: this.fileFilter,
+      ruleidSearchTerms: SearchModule.inFileSearchTerms.ruleid,
+      vulidSearchTerms: SearchModule.inFileSearchTerms.vulid,
+      stigidSearchTerms: SearchModule.inFileSearchTerms.stigid,
+      classificationSearchTerms: SearchModule.inFileSearchTerms.classification,
+      groupNameSearchTerms: SearchModule.inFileSearchTerms.groupName,
+      cciSearchTerms: SearchModule.inFileSearchTerms.cci,
+      titleSearchTerms: SearchModule.inFileSearchTerms.title,
+      descriptionSearchTerms: SearchModule.inFileSearchTerms.description,
+      nistIdFilter: SearchModule.inFileSearchTerms.NISTIdFilter,
+      codeSearchTerms: SearchModule.inFileSearchTerms.code,
+      omit_overlayed_controls: true,
+      control_id: this.controlSelection || undefined,
+      iaControlsSearchTerms: SearchModule.inFileSearchTerms.iaControls,
+      keywordsSearchTerms: SearchModule.inFileSearchTerms.keywords
+    };
+  }
+
+  /**
+   * Clear all filters
+   */
+  clear(clearSearchBar = false) {
+    SearchModule.clear();
+    this.filterSnackbar = false;
+    this.controlSelection = null;
+    if (clearSearchBar) {
+      this.searchTerm = '';
+    }
+  }
+
+  /**
+   * Returns true if we can currently clear.
+   * Essentially, just controls whether the button is available
+   */
+  get canClear(): boolean {
```

Comment:
implement the snackbar on the checklist page so that it warns the user when they've filtered out all their results somehow

if possible review/simplify the implementation for this function esp so the side effect of 'snackbar pops up' is not linked to 'check if we are clearable'

Reply:
Implemented snackbar and added getter to handle enabling and disabling snackbar